src/main/java/ru/testtask/service/ProjectService.java

```diff
@@ -30,13 +38,20 @@ public Project findProjectByName(String name){
     }
 
 
-    public Project createProject(Project project) {
-        for (int i = 0; i < 10; i++) {
-            project.addGeometry(Geometry.builder().name("geometry" + i).id(UUID.randomUUID().toString()).build());
-            project.addAttribute(Attribute.builder().name("geometry" + i).id(UUID.randomUUID().toString()).build());
+    public Project createProject(Project project) throws NameAlreadyExistsException {
+        if (findProjectByName(project.getName()).getId() != null){
+            throw new NameAlreadyExistsException("Project with the same name already exists!");
+        }
+
+        else {
+            for (int i = 0; i < 10; i++) {
+                project.addGeometry(Geometry.builder().name("geometry" + i).id(UUID.randomUUID().toString()).build());
+                project.addAttribute(Attribute.builder().name("geometry" + i).id(UUID.randomUUID().toString()).build());
+            }
+            project.setOwnerId(userService.getCurrentUserId());
+            projectRepo.insert(project);
```

Comment:
`return projectRepo.insert(project);` is better because you will not make an additional request to db